internal/allocator/allocator.go

```diff
@@ -381,3 +372,168 @@ func ipConfusesBuggyFirmwares(ip net.IP) bool {
 	}
 	return ip[3] == 0 || ip[3] == 255
 }
+
+// Dual-stack.
+
+// AssignDual assigns the requested ip's to svc, if the assignment is
+// permissible by sharingKey and backendKey.
+func (a *Allocator) AssignDual(svc string, ip, ip2 net.IP, ports []Port, sharingKey, backendKey string) error {
+	if ipIsIPv6(ip) == ipIsIPv6(ip2) {
+		return fmt.Errorf("%q %q is has the same family", ip, ip2)
+	}
+	pool := poolFor(a.pools, ip)
+	if pool == "" {
+		return fmt.Errorf("%q is not allowed in config", ip)
+	}
+	if poolFor(a.pools, ip2) != pool {
```

Comment:
so for this to work we need to have only one address-pool with both addresses something like 
```
addresses:
  - 2620:52:0:4974:4da7:3546:2a72:3fdf - 2620:52:0:4974:4da7:3546:2a72:3fdf
  - 10.73.116.54 - 10.73.116.54
```
having address-pool for v4 and another for v6 won't work right ?

Reply:
No. If you did not have that restriction you must basically rewrite the whole allocator. Beside IMO it is a natural restriction.

Reply:
ok lets be very specific and clear on this can you layout example of what will work and what won't to be 100% sure we are clear ?